pkg/kubelet/kubelet_node_status.go

```diff
@@ -404,18 +407,29 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 		return fmt.Errorf("nil %q node object", kl.nodeName)
 	}
 
+	nodeHasChanged := false
 	if node.Spec.PodCIDR != "" {
+		nodeHasChanged = true
 		if err := kl.updatePodCIDR(node.Spec.PodCIDR); err != nil {
 			glog.Errorf(err.Error())
 		}
 	}
 
 	kl.setNodeStatus(node)
+
+	now := kl.clock.Now()
+	if utilfeature.DefaultFeatureGate.Enabled(features.NodeLease) && now.Before(kl.lastStatusReportTime.Add(kl.nodeStatusReportFrequency)) {
+		if nodeHasChanged = nodeHasChanged || nodeStatusHasChanged(&originalNode.Status, &node.Status); !nodeHasChanged {
```

Comment:
is the assignment of nodeHasChanged to itself intentional here?

Reply:
Good catch!

Reply:
I think the assignment happens after ||. Anyway, I changed the logic here, so the original if statement is obsolete.